src/search/Search.tsx

```diff
@@ -99,30 +97,30 @@ export function Search({ controller, onClose }: SearchProps) {
             onClick={(event) => {
```

Comment:
moim zdaniem ten onClick powinien wylecieć bo kilknięcie na ten element domu w tym momencie robi wszystko poza interakcją z tym elementem :D
1)jeśli chodzi o wywołanie poszczególnych akcji na kilku to rekomenduje zapiąć onClick w resultsView na poszczególnych


2) jeśli chodzi o zamykanie dialogu na click poza okienkiem to polecam najbardziej reużywany snippet handleClickOutside https://stackoverflow.com/questions/54391682/detect-click-outside-component-react-hooks :)

Reply:
Ad. 1) Czy to jest jakaś ogólnie przyjęta dobra praktyka, żeby eventy na elemencie były obsługiwane w przypadku interakcji z nim? 

Ad. 2) Lol, rozśmieszyło mnie, że nie przyszło mi do głowy takie oczywiste rozwiązanie tylko jakieś fikołki w onClick, dzięki :D

Reply:
Ad. 1) Tak - to jest dobra praktyka, żeby zapinać eventy na elementach triggerujących akcje, ułatwia to organizacje kodu i zrozumienie flow np. mając button "Submit" oczekujesz, że click event hander będzie zapięty na nim :)